### PR TITLE
Get QEMU logs from virtlogd

### LINKED ISSUE #2953: Evacuate QEMU logs from virt-launcher pod
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement

**What happened**:
A virt-launcher pod is being deleted because the QEMU crashed. This happens fast and if "spec.running" is True it happens in loops. Currently there is no option to treat the QEMU logs.

**What you expected to happen**:
Print QEMU logs to the STDOUT and evacuate them to virt-handler for bug tracking.

**How to reproduce it (as minimally and precisely as possible)**:
Run Windows 2019 VM and enable kernel debugging via Network
In order to do so run the following commands as administrator from the command prompt:
bcdedit /debug on
bcdedit /dbgsettings net hostip:192.168.0.12 port:50000 key:this.key.isnt.secure

### PR DESCRIPTION
**Pull request by victortoso**
**What this PR does / why we need it**:
* It reads the logs from /var/log/libvirt/qemu/domain.log (which is handled by virtlogd), structuralizes it and log them with  kubevirt.io/client-go/log so it is reachable outside the pod

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2953 

**Special notes for your reviewer**:
This is a WIP, I want to double check that this is the right direction. I'm just reading from the file created by virtlogd for now.

**Release note**:

```release-note
QEMU logs can now be fetched from outside the pod
```

### CODE REVIEW

**Comment by victortoso** [comment]
CC @enp0s3

File: pkg/virt-launcher/virtwrap/util/libvirt_helper.go [hunk 1]
```diff
@@ -242,10 +242,50 @@ func StartVirtlog(stopChan chan struct{}) {
 
 			err := cmd.Start()
 			if err != nil {
-				log.Log.Reason(err).Error("failed to start libvirtd")
+				log.Log.Reason(err).Error("failed to start virtlogd")
 				panic(err)
 			}
 
+			go func() {
+				logfile := fmt.Sprintf("/var/log/libvirt/qemu/%s.log", domainName)
```
Comment (rmohr) [inline pkg/virt-launcher/virtwrap/util/libvirt_helper.go:250]: I think that it is possible to configure virtlogd to directly print to stdout. Would it be possible to read them directly from virtlogd?

Comment (enp0s3) [reply pkg/virt-launcher/virtwrap/util/libvirt_helper.go:250]: @rmohr @victortoso Following my conversation with Daniel Berrange, the virtlogd can log to stderr only about it's own activity. Qemu logs will continue to be written to /var/log/libvirt/qemu. Even though we configure stdio_handler=logd in qemu.conf this pipe is only between qemu and virtlogd. As per my understanding from Daniel we cannot make virtlogd to redirect the incoming QEMU logging to something other than a file in /var/log/libvirt/qemu

Comment (rmohr) [reply pkg/virt-launcher/virtwrap/util/libvirt_helper.go:250]: Fine by me then. That explains why my configuration attempts in the past had no influence.

File: pkg/virt-launcher/virtwrap/util/libvirt_helper.go [hunk 2]
```diff
@@ -242,10 +242,50 @@ func StartVirtlog(stopChan chan struct{}) {
 
 			err := cmd.Start()
 			if err != nil {
-				log.Log.Reason(err).Error("failed to start libvirtd")
+				log.Log.Reason(err).Error("failed to start virtlogd")
 				panic(err)
 			}
 
+			go func() {
```
Comment (rmohr) [inline pkg/virt-launcher/virtwrap/util/libvirt_helper.go:249]: Can we add some unit tests for this to `libvirt_helper_test.go` which transforms typical qemu logs to kubevirt logs?

File: pkg/virt-launcher/virtwrap/util/libvirt_helper.go (hunk 1 above)
Comment (victortoso) [reply pkg/virt-launcher/virtwrap/util/libvirt_helper.go:250]: Yep, same with me. I was about to send an email to libvirt folks to see what I was doing wrong. Thanks @enp0s3 for the quick check with Daniel.

File: pkg/virt-launcher/virtwrap/util/libvirt_helper.go (hunk 2 above)
Comment (victortoso) [reply pkg/virt-launcher/virtwrap/util/libvirt_helper.go:249]: Sure, I'll be adding that shortly. Thanks!

**Comment by victortoso** [comment]
Besides adding a unit test, I changed the subcomponent to "qemu" as suggested by @enp0s3 and renamed the log function to reflect that.

**Comment by rmohr** [comment]
Let's see what the tests say about the change. @victortoso thanks.

**Pull request merged**